Module-11/Library/Repositories/IDocumentRepository.cs

```diff
@@ -0,0 +1,8 @@
+﻿using Library.Documents;
+
+namespace Library.Repositories;
+
+public interface IDocumentRepository
+{
+    public IEnumerable<IDocument?> GetByNumber(int documentNumber);
```

Comment:
repository must return already aggregated data, so you should not return IEnumerable